helper/schema/schema_test.go

```diff
@@ -3335,6 +3335,25 @@ func TestSchemaMap_Validate(t *testing.T) {
 				fmt.Errorf("\"optional_att\": conflicts with required_att (\"required-val\")"),
 			},
 		},
+
+		"Readonly attribute is set by user": {
+			Schema: map[string]*Schema{
+				"readonly_att": &Schema{
+					Type:     TypeString,
+					ReadOnly: true,
+					Required: true,
+				},
+			},
+
+			Config: map[string]interface{}{
+				"readonly_att": "user-set-val",
+			},
+
+			Err: true,
+			Errors: []error{
+				fmt.Errorf("\"readonly_att\": This field is readonly and cannot be set by users"),
+			},
+		},
```

Comment:
Absolutely. I actually thought I tried that case last night and it worked fine but wasn't quite sure how to test it. I'll add the test in a few moments.